x-pack/plugins/cases/common/utils/markdown_plugins/utils.ts

```diff
@@ -16,14 +16,14 @@ import { SerializableRecord } from '@kbn/utility-types';
 import { LENS_ID, LensParser, LensSerializer } from './lens';
 import { TimelineSerializer, TimelineParser } from './timeline';
 
-interface LensMarkdownNode extends Node {
+export interface LensMarkdownNode extends Node {
   timeRange: TimeRange;
   attributes: SerializableRecord;
   type: string;
   id: string;
 }
 
-interface LensMarkdownParent extends Node {
+export interface MarkdownNode extends Node {
   children: Array<LensMarkdownNode | Node>;
```

Comment:
nit: I think because the children are of type `LensMarkdownNode` and not generic nodes maybe it is more appropriate to name it `LensMarkdownNode` and rename the `LensMarkdownNode` to `LensMarkdownLeaf`

Reply:
If you want to make it generic you can create the following interface:

```
interface MarkdowNode<T> extends Node {
	children: Array<T> | Node;
}

type LensMarkdownNode = MarkdowNode<LensMarkdownLeaf>
```

Reply:
> nit: I think because the children are of type LensMarkdownNode and not generic nodes maybe it is more appropriate to name it LensMarkdownNode and rename the LensMarkdownNode to LensMarkdownLeaf

This function `parseCommentString` returns a node with children of two different types: `children: Array<LensMarkdownNode | Node>;` If the comment is a lens visualization, then the children will likely only be an array of `LensMarkdownNode`. If the comment is just a text string or a timeline link then the `children` will be an array of just `Node`.

The reason I thought `MarkdownNode` was appropriate is because the `parseCommentString` converts the text into a markdown AST in both cases (wether it is a lens viz or normal string comment).

I'll add a comment explaining how it gets converted into markdown.

Reply:
Ok, let's stick with `MarkdownNode` and add more types to the `Array` if we add more plugins.